src/internal/storage/fileset/fileset_test.go

```diff
@@ -291,6 +299,14 @@ func testStableHash(ctx context.Context, t *testing.T, data, expected []byte, ms
 		ids = append(ids, *id)
 	}
 	getHash := func() []byte {
+		if compact {
+			compactFunc := func(ctx context.Context, ids []ID, ttl time.Duration) (*ID, error) {
```

Comment:
This literal seems unnecessary.  Am I missing something?  Can we pass `storage.Compact` instead?

Reply:
No, the signatures are different. `storage.Compact` has functional options.